src/metaxy/metadata_store/delta.py

```diff
@@ -0,0 +1,304 @@
+
+    def _create_system_tables(self) -> None:
+        """Create system tables if they don't exist.
+
+        Creates empty system tables with proper schemas:
+        - metaxy-system__feature_versions: Tracks feature versions and graph snapshots
+        - metaxy-system__migration_events: Tracks migration execution events
+
+        This method is idempotent - safe to call multiple times.
+        """
+        # Check and create feature_versions table
+        feature_versions_path = self._feature_path(FEATURE_VERSIONS_KEY)
+        if not self._table_exists(feature_versions_path):
+            empty_df = pl.DataFrame(schema=FEATURE_VERSIONS_SCHEMA)
+            self._write_metadata_impl(FEATURE_VERSIONS_KEY, empty_df)
+
+        # Check and create migration_events table
+        migration_events_path = self._feature_path(MIGRATION_EVENTS_KEY)
+        if not self._table_exists(migration_events_path):
+            empty_df = pl.DataFrame(schema=MIGRATION_EVENTS_SCHEMA)
+            self._write_metadata_impl(MIGRATION_EVENTS_KEY, empty_df)
+
+    def _table_name_to_feature_key(self, table_name: str) -> FeatureKey:
+        """Convert table name back to feature key.
+
+        Args:
+            table_name: Table name (directory name) to parse
+
+        Returns:
+            FeatureKey constructed from table name parts
+        """
+        # Table names are created by joining parts with "__"
+        parts = table_name.split("__")
+        return FeatureKey(parts)
+
+    def _feature_path(self, feature_key: FeatureKey) -> Path:
+        """Get the filesystem path for a feature's Delta table."""
+        table_name = feature_key.table_name
+        return self.root_path / table_name
+
+    def _table_exists(self, table_path: Path) -> bool:
+        """Check if a Delta table exists at the given path."""
+        return (table_path / "_delta_log").exists()
+
+    # ===== Storage operations =====
+
+    def _write_metadata_impl(
+        self,
+        feature_key: FeatureKey,
+        df: pl.DataFrame,
+    ) -> None:
+        """Append metadata to the Delta table for a feature.
+
+        Args:
+            feature_key: Feature key to write to
+            df: DataFrame with metadata (already validated)
+
+        Raises:
+            TableNotFoundError: If table doesn't exist and auto_create_tables is False
+        """
+        import deltalake  # pyright: ignore[reportMissingImports]
+
+        table_path = self._feature_path(feature_key)
+        table_exists = self._table_exists(table_path)
+
+        # Check if table exists
+        if not table_exists and not self.auto_create_tables:
+            raise TableNotFoundError(
+                f"Delta table does not exist for feature {feature_key.to_string()} at {table_path}. "
+                f"Enable auto_create_tables=True to automatically create tables."
+            )
+
+        # Create parent directory if needed
+        table_path.parent.mkdir(parents=True, exist_ok=True)
```

Comment:
Won't work with object stores, use deltalake native APIs instead

Reply:
see the current approach - it verifies connectivity